### DISCUSSION TITLE
How to generate sitemap.js with i18n routing and a dynamic route (with [slug])?

### DISCUSSION THREAD

**Discussion by mrksph** [amannn/next-intl#1130] [Q&A]
Hi 

First of all I'm still in awe with this library how easy it is to add i18n support and all the features it provides.

I was following through the App Router example and all is working as expected. Now I've added a dynamic route `/projects/[slug]`. I've updated my config.js file to add this new route to the pathnames object 
```
export const pathnames = {
    '/': '/',
    '/contact': {
	es:'/contacto',
	en:'/contact'
    },
    '/projects/[slug]': {
        es: '/proyectos/[slug]',
        en: '/projects/[slug]'
    }
}
```
And now I'd like to generate the sitemap.xml to include all projects under /projects route. 

```
import { defaultLocale, host, locales, pathnames } from "@/navigation/config";
import { getPathname } from "@/navigation/navigation";

const getUrl = (key, locale) => {
	let pathname = getPathname({ locale, href: key });
	return `${host}/${locale}${pathname === '/' ? '' : pathname}`;
};

export default function sitemap() {
	const allPathnames = Object.keys(pathnames);

	const staticPages = allPathnames.map((key) => ({
		url: getUrl(key, defaultLocale),
		lastModified: new Date().toISOString(),
		priority: 1,
		changeFrequency: 'monthly'
	}));

	return [...staticPages];
}
```

But I'm getting this error:

```
⨯ Error: Insufficient params provided for localized pathname.
Template: /proyectos/[slug]
```

I've already checked the docs for the getPathname() function ([https://next-intl-docs.vercel.app/docs/routing/navigation#getpathname](https://next-intl-docs.vercel.app/docs/routing/navigation#getpathname)) and If I pass a single slug it compiles and generates the sitemap.xml correctly but I don't know how to either a) filter out all routes that have a [slug] sub-segment b) if it's correct to do it this way c) there's a way to provide all valid slugs to the params config so I can generate the sitemap.xml.

**Comment by mndgs1** [comment]
Hey, did you find a fix for this? I am currently having the same problem..

**Comment by SamuelChojnacki** [comment]
same probleme here !!!! i use a script in build time for my part

**Comment by amannn** [comment]
This is covered in the docs: [Sitemap](https://next-intl-docs.vercel.app/docs/environments/actions-metadata-route-handlers#sitemap)
